core/sql/comexe/ComTdbExeUtil.cpp

```diff
@@ -1302,6 +1302,7 @@ void ComTdbExeUtilFastDelete::displayContents(Space * space,
 ComTdbExeUtilHiveTruncate::ComTdbExeUtilHiveTruncate(
      char * tableName,
      ULng32 tableNameLen,
+     char * hiveTableName,
```

Comment:
I think I see the reason for the confusion. We are using an existing operator that until now referred to Hive tables specified in the SQL statement itself. For those purposes, binding at compile time makes sense. I'm wondering though if for temporary tables whose data is used only within the scope of one execution (such as the ones you're introducing with CSE) should have their names bound at run time instead. So the ComTdbExeUtilHiveTruncate tdb would have to be extended to have two modes: One where the table name is passed to it by the generator and one where the table name is missing and some other token or mechanism to identify it is used instead (e.g. "temporary table number 3"). In the latter case the corresponding TCB would perhaps consult an executor global data structure to learn the table name.